target_chains/solana/sdk/js/pyth_solana_receiver/src/vaa.ts

```diff
@@ -160,3 +166,185 @@ export async function findEncodedVaaAccountsByWriteAuthority(
   });
   return result.map((account) => new PublicKey(account.pubkey));
 }
+
+/**
+ * Build a set of instructions to write two VAAs (start and end) to encoded VAA accounts.
+ *
+ * The instructions are assembled in an opinionated way so that they can be efficiently packed in 3 transactions:
+ * TX 1. Create, init and write initial data for start VAA
+ * TX 2. Create, init and write initial data for end VAA
+ * TX 3. Write remaining data and verify start and end VAAs
+ *
+ * @param wormhole The Wormhole program instance
+ * @param startUpdateData The accumulator update data for the start VAA
+ * @param endUpdateData The accumulator update data for the end VAA
+ * @returns Instructions for posting VAAs, closing VAAs, and the VAA data
+ */
+export async function buildPostEncodedVaasForTwapInstructions(
+  wormhole: Program<WormholeCoreBridgeSolana>,
+  startUpdateData: AccumulatorUpdateData,
+  endUpdateData: AccumulatorUpdateData
+): Promise<{
+  postInstructions: InstructionWithEphemeralSigners[];
+  closeInstructions: InstructionWithEphemeralSigners[];
+  startVaa: {
+    trimmedVaa: Buffer;
```

Comment:
I wouldn't used trimmedVaa here since these are fully verified VAAs

Reply:
I think you only need to return encodedVaa as a Pubkey like in `buildPostEncodedVaaInstructions`